materialize-firebolt/transactor.go

```diff
@@ -163,16 +163,31 @@ func (t *transactor) projectDocument(spec *pf.MaterializationSpec_Binding, keys
 
 	// Add the keys to the document.
 	for i, value := range keys {
-		var propName = spec.FieldSelection.Keys[i]
+		var propName = strings.ReplaceAll(spec.FieldSelection.Keys[i], "/", "_")
```

Comment:
There might be a better way of approaching this. Some background:

The `FieldSelection` represents the set of projections that should be materialized. During `flowctl api build`, assuming the user has left the default [`fields`](https://github.com/estuary/flow/blob/master/crates/models/src/materializations.rs#L49) in their binding, we'll call the `Validate` RPC of the connector and resolve this `FieldSelection` by just taking all the fields that the connector tells us are "recommended".

So I think a better way of handling this would be to validate the field names as part of the `Validate` RPC and return `FIELD_FORBIDDEN` for any that contain a `/`. If any fields would need to be renamed, then the expectation would be that the user provides an explicit projection and uses the `fields` configuration in the materialization binding to wire it up. It's more work for a CLI user, but it allows the `projections` of a collection to remain authoritative on the mappings between tabular and JSON representations, and can be guided by good `reason` strings in the constraints returned by `Validate`. It also avoids the possibility of collisions in the case where there's both `foo/bar` and `foo_bar` fields.

Reply:
@psFried I see, okay, I thought since the default recommended fields seem to contain `/`, it would be simple if we just allow the recommendations, only with a change in the field name. This means our recommended fields will not be valid for any nested objects and any nested objects will have to be manually projected, right? Is there a better way where we can still allow the default recommendations?

Reply:
Authoritativeness of the projections and avoiding implicit projections makes sense though, I see the point.

Reply:
I'm looking to see if there is a way in Firebolt to allow column names with `/` in them (perhaps by some form of quotation or escaping)

Reply:
@psFried added validation logic for column names and no longer manipulating them in the connector / schemalate: https://github.com/estuary/flow/pull/423/commits/e24b9685267c021792b8444fb3da39d949ed05d9

Reply:
The result is:
```
constraints:<key:\"obj/name\" value:<type:FIELD_FORBIDDEN reason:\"Field names must match the regex ^[a-zA-Z_][a-zA-Z0-9_]*$.\" >
```

Which leads to `obj/name` being omitted from the document and the table.

Reply:
👍 to resolution here.

>  I get a 500 Internal Error from Firebolt if I try to create a column with / in it, e.g.: CREATE FACT TABLE ("obj/name" TEXT) PRIMARY INDEX "obj/name" fails with 500.

This smells like something they ought to be able to fix -- that in the future "obj/name" would be fine with proper quoting.